spec/components/hiring_staff/vacancy_form_page_heading_component_spec.rb

```diff
@@ -0,0 +1,125 @@
+require "rails_helper"
+
+RSpec.describe HiringStaff::VacancyFormPageHeadingComponent, type: :component do
+  let(:organisation) { create(:school, name: "Teaching Vacancies Academy") }
+  let(:state) { "create" }
+  let(:vacancy) { create(:vacancy, state: state, job_title: "Test job title") }
+  let(:session_job_location) { nil }
+  let(:session_readable_job_location) { nil }
+  let(:session_vacancy_attributes) { { "job_location" => session_job_location, "readable_job_location" => session_readable_job_location } }
+  let(:current_step) { 99 }
+
+  subject { described_class.new(vacancy, session_vacancy_attributes) }
+
+  before do
+    allow(subject).to receive(:current_step).and_return(current_step)
+    allow(subject).to receive(:total_steps).and_return(100)
+    allow(subject).to receive(:current_organisation).and_return(organisation)
+    vacancy.organisation_vacancies.create(organisation: organisation) if vacancy.present?
+    render_inline(subject)
+  end
+
+  describe "#show_current_step?" do
+    context "when vacancy state is create or review" do
+      let(:state) { "create" }
+
+      it "shows the current step" do
+        expect(rendered_component).to include(I18n.t("jobs.current_step", step: current_step, total: 100))
+      end
+    end
+
+    context "when vacancy state is not create or review" do
+      let(:state) { "copy" }
+
+      it "does not show the current step" do
+        expect(rendered_component).not_to include(I18n.t("jobs.current_step", step: 100, total: 101))
```

Comment:
I wonder if being less specific here about what not to include might be a more revealing test?

Reply:
Either way wouldn't we want it to not include `step: current_step`?

Reply:
I'll change it to `expect(rendered_component.downcase).not_to include('step')` - ok?

Reply:
I prefer the first